### ISSUE TITLE
[BUG] `save()` method fails to save a trained `CNNClassifier` instance

### ISSUE THREAD

**Issue by tapyu** [sktime/sktime#6275]
**Describe the bug**
If

```py
cnn = CNNClassifier(n_epochs=100, batch_size=4)
```

and if `cnn.fit()` has been run, the command

```
cnn.save("save_it")
```

Fails and prompts the following warning

> Invalid filepath extension for saving. Please add either a `.keras` extension for the native Keras format (recommended) or a `.h5` extension. Use `tf.saved_model.save()` if you want to export a SavedModel for use with TFLite/TFServing/etc. Received: filepath=save_it/keras.

**To Reproduce**

1. Get any training dataset, `X_trn` with its respective labels, `y_trn`
1. Run
```python
from sktime.classification.deep_learning.cnn import CNNClassifier
cnn = CNNClassifier(n_epochs=100, batch_size=4)
cnn.fit(X_trn, y_trn)
cnn.save("save_it")
```

**Expected behavior**
The trained model should be saved.

**Additional context**
- If `cnn.save()` has not been run. The model is saved. However, from the [documentation](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.classification.deep_learning.CNNClassifier.html#sktime.classification.deep_learning.CNNClassifier.save), it is not clear that the [`save()`](https://github.com/sktime/sktime/blob/43171265241f41998ec90665234e24120ec40332/sktime/classification/deep_learning/base.py#L208-L307) function must be used only for untrained instances.
- adding `serialization_format='pickle'` changes nothing.

**Versions**
```
>>> from sktime import show_versions; show_versions()
/home/tapyu/.cache/pypoetry/virtualenvs/tscnn-zfTdj8bO-py3.10/lib/python3.10/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

System:
    python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
executable: /home/tapyu/.cache/pypoetry/virtualenvs/tscnn-zfTdj8bO-py3.10/bin/python
   machine: Linux-6.5.0-26-generic-x86_64-with-glibc2.35

Python dependencies:
          pip: 23.3.1
       sktime: 0.25.1
      sklearn: 1.3.2
       skbase: 0.7.1
        numpy: 1.26.3
        scipy: 1.12.0
       pandas: 2.1.4
   matplotlib: 3.8.2
       joblib: 1.3.2
        numba: None
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: 2.16.1
tensorflow_probability: None
```

</details>

**Comment by fkiraly** [comment]
I believe this is an instance of this bug which is an incompatibility between more recent versions of `tensorflos` and `sktime` serialization logic for deep learning classifiers:
https://github.com/sktime/sktime/issues/6094

The bugfix is present since 0.27.1, so what should fix it for you is either downgrading `tensorflow`, or upgrading `sktime`.

Please let us know if you think this is a different bug.

**Comment by tapyu** [comment]
Yes, just to let you now, I upgraded `sktime` from `0.25.1` to `0.28.0` and this problem disappeared.
